r1cs/gadgets/std/src/multieq.rs

```diff
@@ -0,0 +1,127 @@
+use algebra::{PrimeField, FpParameters};
```

Comment:
Actually, we don't need this multieq at all (Please note that you are not using its functions anyway). If you look in Bellman implementation, it is only used in UInt32::addmany() as a transparent way to accumulate the bits into linear combinations and then enforcing the equality between them.
Our code does the same without using this MultiEq ConstraintSystem. Look at the diff between our (left side) addmany and Zcash one (right side):

https://www.diffchecker.com/pFPqK7Hd